api/v1beta1/flowcollector_webhook.go

```diff
@@ -16,17 +16,134 @@ limitations under the License.
-func (r *FlowCollector) SetupWebhookWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
-		Complete()
+	"github.com/netobserv/network-observability-operator/api/v1beta2"
+	utilconversion "github.com/netobserv/network-observability-operator/pkg/conversion"
+	"github.com/netobserv/network-observability-operator/pkg/helper"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this v1beta1 FlowCollector to its v1beta2 equivalent (the conversion Hub)
+// https://book.kubebuilder.io/multiversion-tutorial/conversion.html
+func (r *FlowCollector) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta2.FlowCollector)
+
+	if err := Convert_v1beta1_FlowCollector_To_v1beta2_FlowCollector(r, dst, nil); err != nil {
+		return fmt.Errorf("copying v1beta1.FlowCollector into v1beta2.FlowCollector: %w", err)
+	}
+	dst.Status.Conditions = make([]v1.Condition, len(r.Status.Conditions))
+	copy(dst.Status.Conditions, r.Status.Conditions)
+
+	// Manually restore data.
+	restored := &v1beta2.FlowCollector{}
+	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
+		// fallback on current loki config as Manual mode if metadata are not available
+		dst.Spec.Loki.Mode = v1beta2.LokiModeManual
+		dst.Spec.Loki.Manual.IngesterURL = r.Spec.Loki.URL
+		dst.Spec.Loki.Manual.QuerierURL = r.Spec.Loki.QuerierURL
+		dst.Spec.Loki.Manual.StatusURL = r.Spec.Loki.StatusURL
+		dst.Spec.Loki.Manual.TenantID = r.Spec.Loki.TenantID
+		dst.Spec.Loki.Manual.AuthToken = r.Spec.Loki.AuthToken
+		if err := Convert_v1beta1_ClientTLS_To_v1beta2_ClientTLS(&r.Spec.Loki.TLS, &dst.Spec.Loki.Manual.TLS, nil); err != nil {
+			return fmt.Errorf("copying v1beta1.Loki.TLS into v1beta2.Loki.Manual.TLS: %w", err)
+		}
+		if err := Convert_v1beta1_ClientTLS_To_v1beta2_ClientTLS(&r.Spec.Loki.StatusTLS, &dst.Spec.Loki.Manual.StatusTLS, nil); err != nil {
+			return fmt.Errorf("copying v1beta1.Loki.StatusTLS into v1beta2.Loki.Manual.StatusTLS: %w", err)
+		}
+		return err
+	}
+
+	// Processor
+	dst.Spec.Processor.LogTypes = restored.Spec.Processor.LogTypes
+	if restored.Spec.Processor.ConversationHeartbeatInterval != nil {
+		dst.Spec.Processor.ConversationHeartbeatInterval = restored.Spec.Processor.ConversationHeartbeatInterval
+	}
+	if restored.Spec.Processor.ConversationEndTimeout != nil {
+		dst.Spec.Processor.ConversationEndTimeout = restored.Spec.Processor.ConversationEndTimeout
+	}
+	if restored.Spec.Processor.Metrics.DisableAlerts != nil {
+		dst.Spec.Processor.Metrics.DisableAlerts = restored.Spec.Processor.Metrics.DisableAlerts
+	}
```

Comment:
these fields were already present in v1beta1, I don't think this conversion is really needed?

Reply:
that's right, I missed to remove that part. Thanks